centrallix/tests/t_driver.c

```diff
@@ -25,23 +25,31 @@
 /************************************************************************/
 
 
-long long test(void);
+#define RED    "\033[1m\x1B[31m"
+#define GREEN  "\033[1m\x1B[32m"
+#define RESET  "\x1B[0m"
+
+long long test(char**);
+
+char * test_name = "?";
 
 void
 segv_handler(int v)
     {
-    exit(121);
+    printf(RESET "%-62.62s  " RED "CRASH\n" RESET, test_name);
+    exit(0);
     }
 void
 abort_handler(int v)
     {
-    printf("FAIL\n");
+    printf(RESET "%-62.62s  " RED "ABORT\n" RESET, test_name);
```

Comment:
Is there a semantic difference between test() returning nonzero and assert() causing SIGABRT?  My sense is that these should be treated similarly in the output (both as FAIL).  The coder can still tell what happened if assert() is called, via its output.